src/main/java/games/strategy/engine/config/PropertyFileReader.java

```diff
@@ -27,14 +27,22 @@
 
 
   String readProperty(final String propertyKey) {
+    return readProperty(propertyKey, true);
+  }
+
+  String readProperty(final String propertyKey, boolean throwException) {
```

Comment:
See the PR after this one for how it would evolve: https://github.com/triplea-game/triplea/pull/1901/files#diff-0c9333ae5c05bfe9e3d83f3c09c9e077

I found that requiring properties was a bit constraining. The approach I went with was returning empty string for not found case. Clients know if they require a value or not and can throw an exception. For example, even with this update, a required value may not be set, and so clients have to check for the empty string case anyways.

I do recommend going that route. The boolean parameter is a bit nasty, it violates SRP and encapsulation. You have the client telling this module how it should behave. We also have variance on the return behavior based on parameter, rather than a single return methodology. Returning empty string on the not found case is pretty straight forward, particularly from a client point of view.

`Optional<String>` is something that is really to be avoided. It's pretty rare where empty string is actually different from `Optional.empty()` (aka: null)